src/core/index.rs

```diff
@@ -147,36 +259,34 @@ impl Index {
     /// For other unit tests, prefer the `RAMDirectory`, see: `create_in_ram`.
     #[cfg(feature = "mmap")]
     pub fn create_from_tempdir(schema: Schema) -> crate::Result<Index> {
-        let mmap_directory = MmapDirectory::create_from_tempdir()?;
-        Index::create(mmap_directory, schema)
+        IndexBuilder::new().schema(schema).create_from_tempdir()
     }
 
     /// Creates a new index given an implementation of the trait `Directory`.
     ///
     /// If a directory previously existed, it will be erased.
-    pub fn create<Dir: Directory>(dir: Dir, schema: Schema) -> crate::Result<Index> {
+    pub fn create<Dir: Directory>(
```

Comment:
can we move this guy as a private function of the Builder, and keep the original create for backward compatibility.

Reply:
yes that's better. done

Reply:
I do not see the change. Is it fixed?

Reply:
Yes, the last commit: https://github.com/tantivy-search/tantivy/pull/1021/commits/b9b9e9e518178ca07ce79a2bbd0cde5ba45064ab